File: OOP with JavaScript Code Files/Lesson-1/Code17.cs
//C#

public class Person {
	public string Name {get; set;}
	public string SurName {get; set;}
}

public class Programmer:Person {
	public String KnownLanguage {get; set;}
}

public void WriteFullName(Person p) {
	Console.WriteLine(p.Name + " " + p.SurName);   
}

var a = new Person();
a.Name = "John";
a.SurName = "Smith";

var b = new Programmer();
b.Name = "Mario";
b.SurName = "Rossi";
b.KnownLanguage = "C#";

WriteFullName(a);	//result: John Smith
WriteFullName(b);	//result: Mario Rossi

File: OOP with JavaScript Code Files/Lesson-3/Code17.js
function Person(name, surname) {
  this.name = name;
  this.surname = surname;
}

var myMixin = {
  getFullName: function() { 
	return this.name + " " + this.surname;
  }
};

function augment(destination, source) {
  for (var methodName in source) {
	if (source.hasOwnProperty(methodName)) {
      destination[methodName] = source[methodName];
	}
  }
  return destination; 
}

augment(Person.prototype, myMixin);

var johnSmith = new Person("John", "Smith");

console.log(johnSmith.getFullName());		//result: "John Smith"
